bin/ansible-connection

```diff
@@ -189,69 +140,112 @@ class Server():
 
                     signal.alarm(self.play_context.timeout)
```

Comment:
Also, it might be useful to setup a different alarm handler for this alarm to log that it was a command timeout that caused the socket to be interrupted